compiler/src/iree/compiler/Codegen/Common/EncodingInfo.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_SRC_IREE_COMPILER_CODEGEN_COMMON_ENCODINGINFO_H_
+#define IREE_COMPILER_SRC_IREE_COMPILER_CODEGEN_COMMON_ENCODINGINFO_H_
+
+#include "iree-dialects/Dialect/LinalgExt/Passes/Passes.h"
+#include "iree/compiler/Dialect/HAL/IR/HALTypes.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+enum class MatmulType {
+  F32F32F32,
+  I8I8I32,
+};
+
+enum class MatmulOperandRole {
+  LHS,
+  RHS,
+  RHS_TRANSPOSE,
+  RESULT,
+};
+
+struct MatmulTileParams {
+  int64_t M = 1;
+  int64_t K = 1;
+  int64_t N = 1;
+};
+
+/// Extracts encoding from the `tensorType` if specified.
+Optional<IREE::LinalgExt::TensorEncoding> getEncoding(
+    RankedTensorType tensorType);
+
+Optional<MatmulType> getMatmulType(IREE::LinalgExt::TensorEncoding encoding);
+
+Optional<MatmulOperandRole> getMatmulOperandRole(
+    IREE::LinalgExt::TensorEncoding encoding);
+
+void adjustTileSizesToNarrowStaticShape(
+    IREE::LinalgExt::MaterializeEncodingInfo &encodingInfo,
+    ArrayRef<int64_t> shape);
+
+IREE::LinalgExt::MaterializeEncodingInfo chooseEncodingInfoForMatmul(
+    MatmulType type, MatmulOperandRole operandRole,
+    MatmulTileParams tileParams);
+
+IREE::LinalgExt::MaterializeEncodingValueFn getMaterializeEncodingValueFn(
+    IREE::HAL::ExecutableTargetAttr targetAttr);
```

Comment:
TileAndDistribute needs this, which is bad. It makes VMVX/ depends on Common/. This should be removed after TileAndDistribute does not need the method.

Reply:
It might take moving everything to scf.foreach_thread to finally break this dependency. That might be a few weeks though...